client/request/select.go

```diff
@@ -111,25 +108,20 @@ func (s *Select) validateGroupBy() []error {
 }
 
 // selectJson is a private object used for handling json deserialization
-// of `Select` objects.
+// of [Select] objects.
+//
+// It contains everything minus the [ChildSelect], which uses a custom UnmarshalJSON
+// and is skipped over when embedding due to the way the std lib json pkg works.
 type selectJson struct {
 	Field
-	DocIDs      immutable.Option[[]string]
-	CID         immutable.Option[string]
-	Root        SelectionType
-	Limit       immutable.Option[uint64]
-	Offset      immutable.Option[uint64]
-	OrderBy     immutable.Option[OrderBy]
-	GroupBy     immutable.Option[GroupBy]
-	Filter      immutable.Option[Filter]
+	Limitable
+	Offsetable
+	Orderable
+	Filterable
+	DocIDsFilter
+	CIDFilter
+	Groupable
 	ShowDeleted bool
-
-	// Properties above this line match the `Select` object and
-	// are deserialized using the normal/default logic.
-	// Properties below this line require custom logic in `UnmarshalJSON`
-	// in order to be deserialized correctly.
-
-	Fields []map[string]json.RawMessage
 }
 
 func (s *Select) UnmarshalJSON(bytes []byte) error {
```

Comment:
thought: Since `ChildSelect` has a custom `Unmarshaller`, this should now be unecessary.

Reply:
Was documented in the commit message:

> The use of a ChildSelect object allows the custom code for desrialising fields to live on a dedicated object. Sadly, due to a very annoying 'quirk' of the json lib, embedded UnmarshalJSON functions are not picked up, and so we still have to do some manual property mapping of the parent.

Reply:
It does pick it up. I added print statements in the ChildSelect unmarshal method and it does get in there. Something else is behaving weirdly though.

Reply:
So it's not that the emvbedded UnmarshalJSON is not picked up. It's that it ignores the other fields: https://stackoverflow.com/a/38982567

Reply:
Either way, you have to keep it like it is now. All good 👍.